File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st  = []
        map = {'(':')','{':'}','[':']'}
        for i in s :
            if i in map :
                print(i)
                st.append(i)
            elif len(st) != 0 and map.get(st[-1]) == i :
                st.pop()
            else:
                return False
        return len(st) == 0
                
        
        

File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack();
        for ( int i = 0; i < s.length(); i++) {
            if( s.charAt(i) == '(' || s.charAt(i) =='{' || s.charAt(i) == '[') {
                
                st.push(s.charAt(i));
                System.out.println(s.charAt(i)+"is pushed");
            }
            else {
                switch (s.charAt(i)) {
                    case ')':
                        
                        if(( st.isEmpty() || st.pop() != '('))
                            return false;
                        break;
                    case '}':
                        if((st.isEmpty() || st.pop() != '{'))
                        {
                            System.out.println("breaking here at 18");
                            return false;
                        }
                        break;
                    case ']':
                        if((st.isEmpty() || st.pop() != '['))
                            return false;
                        break;
                    default:
                        return false;
                }  
            }
        }
        if(st.isEmpty())
        {
            System.out.println("is empty");
            return true;
        }
        return false;
    }
}